stream-compositions/services/product-composition-service/src/main/java/com/backbase/stream/compositions/product/core/service/impl/ProductIngestionServiceImpl.java

```diff
@@ -103,6 +97,14 @@ private Mono<ProductIngestResponse> sendToDbs(ProductIngestResponse res) {
                 .map(BatchProductGroupTask::getData)
                 .map(pg -> ProductIngestResponse.builder()
                         .productGroups(pg.getProductGroups().stream().map(g -> (ProductGroup) g).collect(Collectors.toList()))
+                        .serviceAgreementExternalId(res.getServiceAgreementExternalId())
```

Comment:
shouldn't we put this into a mapper?

Reply:
I see that we are actually creating a new `ProductIngestionResponse` from `BatchProductGroup` & then we need to copy IDs from previously created `ProductIngestResponse` .
So, even if I create a new mapper, It will need 2 inputs to fully build the object for `ProductIngestResponse` . One input for `BatchProductGroup`  & other for `ProductIngestResponse` .
Or I might need to create afterMapping or something like that.


But I don’t see stream-service using any similar pattern. So avoided to create unnecessary mapper just to carry forward 3-4 fields.